### PR TITLE
Login: Move the login component out of the auth section

### LINKED ISSUE #12668: Refactor OAuth login in `client/auth`
https://github.com/Automattic/wp-calypso/pull/12594 has introduced a new native login component (native in the sense that it now lives in calypso rather than served by wpcom). We named it `wp-login` as to not conflict it with the oauth component at `client/oauth/login.jsx`.

Given that we wanted to reuse the `/login` route without introducing too many changes in a single PR, it is necessary to have a follow up PR which will refactor `client/auth`. As stated on the PR, we should:

- Mount the OAuth route on something else than `/login`
- Move the current `client/auth/login` component to another place (section?) or merge it with our new component.
- Test that it doesn't introduce breaking changes on `wp-desktop` which uses this oauth component as its splash screen.

### PR DESCRIPTION
**Pull request by scruffian**
This pull request fixes #12668 by moving the new login work to a different section, making it easier to keep exceptions in place for the auth section (for example no masterbar).

#### Testing instructions
  
1. Run `git checkout move/login` and start your server
2. Open the [`Login` page](http://calypso.localhost:3000/login)
3. Check that it loads with a masterbar at the top
4. Update `config/development.json`, setting `features.oauth` to true
5. Open the [`Oauth Login` page](http://calypso.localhost:3000/oauth-login)
6. Check that the desktop app login loads correct.
  
#### Additional notes
  
This will probably require changes to the desktop app, but I haven't tested it yet.
@jblz This might also break magic links.
  
#### Reviews
  
- [ ] Code
- [ ] Product

   
@Automattic/sdev-feed

### CODE REVIEW

**Comment by jblz** [comment]
Looks like `assets/stylesheets/_components.scss` was left out of the branch...

> Error: File to import not found or unreadable: auth/wp-login/style

I have to do this to get it to build:

```
diff --git a/assets/stylesheets/_components.scss b/assets/stylesheets/_components.scss
index 471b628e2..841b1526d 100644
--- a/assets/stylesheets/_components.scss
+++ b/assets/stylesheets/_components.scss
@@ -17,7 +17,6 @@
 @import 'account-recovery/reset-password/transaction-id-form/style';
 @import 'account-recovery/style';
 @import 'auth/style';
-@import 'auth/wp-login/style';
 @import 'blocks/app-promo/style';
 @import 'blocks/author-compact-profile/style';
 @import 'blocks/author-selector/style';
@@ -207,6 +206,7 @@
 @import 'lib/directly/style';
 @import 'lib/keyboard-shortcuts/style';
 @import 'lib/abtest/test-helper/style';
+@import 'login/wp-login/style';
 @import 'mailing-lists/style';
 @import 'me/account-password/style';
 @import 'me/account/style';
```

**Comment by jblz** [comment]
> This might also break magic links.

Magic Links seem to be unaffected by this change. Thanks for the ping (& for tidying this up)!

**Comment by scruffian** [comment]
This has been rebased, and the issues @jblz mentioned fixed.

**Comment by scruffian** [comment]
I tried running this in wp-deskop, but I see this error:

```TypeError: _extensions2.default.reducers is not a function
    at Object.module.exports.Object.defineProperty.value (/Users/ben/Sites/wp-desktop/build/desktop.js:94352:68)
    at __webpack_require__ (/Users/ben/Sites/wp-desktop/build/desktop.js:21:30)
    at Object.<anonymous> (/Users/ben/Sites/wp-desktop/build/desktop.js:110378:15)
    at Object.<anonymous> (/Users/ben/Sites/wp-desktop/build/desktop.js:110756:31)
    at __webpack_require__ (/Users/ben/Sites/wp-desktop/build/desktop.js:21:30)
    at Object.<anonymous> (/Users/ben/Sites/wp-desktop/build/desktop.js:195275:14)
    at Object.<anonymous> (/Users/ben/Sites/wp-desktop/build/desktop.js:195350:31)
    at __webpack_require__ (/Users/ben/Sites/wp-desktop/build/desktop.js:21:30)
    at startServer (/Users/ben/Sites/wp-desktop/build/desktop.js:197247:14)
    at /Users/ben/Sites/wp-desktop/build/desktop.js:197274:5

System info: {"platform":"darwin","release":"16.5.0","version":"2.3.0-beta2","build":"development"}```

**Comment by jblz** [comment]
This all tests fine for me -- I added a "Needs Rebase" label because of the recent node version upgrade, but I think it's fine to merge and figure out the wp-desktop issue(s) separately.

**Comment by scruffian** [comment]
Thanks for finding and fixing the linting issue

File: client/auth/test/login.jsx [hunk 1]
```diff
@@ -1,78 +0,0 @@
-/**
```
Comment (jblz) [inline client/auth/test/login.jsx]: One question -- it looks like this test file didn't make it anywhere else. Intended?

Comment (scruffian) [inline client/auth/test/login.jsx]: Good catch thanks

**Pull request merged**